airbyte-ci/connectors_ci/metadata_service/orchestrator/orchestrator/assets/catalog_assets.py

```diff
@@ -0,0 +1,157 @@
+    all_destinations_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+
+    markdown = "# Connector Catalog Locations\n\n"
+    markdown += "## Sources\n"
+    markdown += all_sources_dataframe[columns_to_show].to_markdown()
+    markdown += "\n\n## Destinations\n"
+    markdown += all_destinations_dataframe[columns_to_show].to_markdown()
+
+    catalog_report_directory_manager = context.resources.catalog_report_directory_manager
+    file_handle = catalog_report_directory_manager.write_data(markdown.encode(), ext="md", key="connector_catalog_locations")
+
+    metadata = {
+        "preview": MetadataValue.md(markdown),
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
+    }
+    return Output(metadata=metadata, value=file_handle)
+
+
+@asset
+def all_destinations_dataframe(cloud_destinations_dataframe, oss_destinations_dataframe) -> pd.DataFrame:
+    """
+    Merge the cloud and oss destinations catalogs into a single dataframe.
+    """
+
+    # Add a column 'is_cloud' to indicate if an image/version pair is in the cloud catalog
+    cloud_destinations_dataframe["is_cloud"] = True
+
+    # Add a column 'is_oss' to indicate if an image/version pair is in the oss catalog
+    oss_destinations_dataframe["is_oss"] = True
+
+    composite_key = ["destinationDefinitionId", "dockerRepository", "dockerImageTag"]
+
+    # Merge the two catalogs on the 'image' and 'version' columns, keeping only the unique pairs
+    merged_catalog = pd.merge(
+        cloud_destinations_dataframe, oss_destinations_dataframe, how="outer", on=composite_key
+    ).drop_duplicates(subset=composite_key)
+
+    # Replace NaN values in the 'is_cloud' and 'is_oss' columns with False
+    merged_catalog[["is_cloud", "is_oss"]] = merged_catalog[["is_cloud", "is_oss"]].fillna(False)
+
+    # Return the merged catalog with the desired columns
+    return merged_catalog
+
+
+@asset
+def all_sources_dataframe(cloud_sources_dataframe, oss_sources_dataframe) -> pd.DataFrame:
+    """
+    Merge the cloud and oss source catalogs into a single dataframe.
+    """
+
+    # Add a column 'is_cloud' to indicate if an image/version pair is in the cloud catalog
+    cloud_sources_dataframe["is_cloud"] = True
+
+    # Add a column 'is_oss' to indicate if an image/version pair is in the oss catalog
+    oss_sources_dataframe["is_oss"] = True
+
+    composite_key = ["sourceDefinitionId", "dockerRepository", "dockerImageTag"]
+
+    # Merge the two catalogs on the 'image' and 'version' columns, keeping only the unique pairs
+    merged_catalog = pd.merge(
+        cloud_sources_dataframe, oss_sources_dataframe, how="outer", on=composite_key
+    ).drop_duplicates(subset=composite_key)
+
+    # Replace NaN values in the 'is_cloud' and 'is_oss' columns with False
+    merged_catalog[["is_cloud", "is_oss"]] = merged_catalog[["is_cloud", "is_oss"]].fillna(False)
+
+    # Return the merged catalog with the desired columns
+    return merged_catalog
```

Comment:
We have similar logic in the QA engine to build this kind of dataframe.
I think it'll be great to migrate the QA engine to Dagster so that it can use these assets. 
The QA engine should definitely be a  Dagster job. What could be a migration strategy to move the QA engine to dagster? Should we consider the QA engine a part of the metadata service? If not how can we share these assets?
@evantahler I'd love to plan "QA engine orchestrated with Dagster" in Q2. I'm a bit worried that we build up the tech debt if we keep it as a separate, GHA-powered, project.

Reply:
+1 on merging these projects.  See comment above